src/serializer/ResidualHeapSerializer.js

```diff
@@ -695,12 +695,25 @@ export class ResidualHeapSerializer {
     let result = new Set(initialGenerators);
     let activeFunctions = functionValues.slice();
     let visitedFunctions = new Set();
-    while (activeFunctions.length > 0) {
+
+    mainLoop: while (activeFunctions.length > 0) {
       let f = activeFunctions.pop();
       if (visitedFunctions.has(f)) continue;
       visitedFunctions.add(f);
+
+      // If this is an additional function, ensure its not
+      // a nested value of any of the function values
+      if (this.optimizedFunctionGenerators.has(f)) {
+        for (let functionValue of functionValues) {
+          if (functionValue !== f && this.optimizedFunctionGenerators.has(functionValue)) {
+            if (this.isNestedOptimizedFunction(f, functionValue)) {
+              continue mainLoop;
```

Comment:
What does this do? Removing this line doesn't seem to fail any tests.

Reply:
Removed, it's no longer needed. I had a test case that I thought needed this but I fixed this in another place.